include/seqan3/alignment/pairwise/detail/policy_affine_gap_recursion_banded.hpp

```diff
@@ -0,0 +1,103 @@
+    using typename base_policy_t::score_type;
+    using typename base_policy_t::affine_cell_type;
+
+    //Import member types.
+    using base_policy_t::gap_extension_score;
+    using base_policy_t::gap_open_score;
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    policy_affine_gap_recursion_banded() = default; //!< Defaulted.
+    policy_affine_gap_recursion_banded(policy_affine_gap_recursion_banded const &) = default; //!< Defaulted.
+    policy_affine_gap_recursion_banded(policy_affine_gap_recursion_banded &&) = default; //!< Defaulted.
+    policy_affine_gap_recursion_banded & operator=(policy_affine_gap_recursion_banded const &) = default; //!< Defaulted.
+    policy_affine_gap_recursion_banded & operator=(policy_affine_gap_recursion_banded &&) = default; //!< Defaulted.
+    ~policy_affine_gap_recursion_banded() = default; //!< Defaulted.
+
+    /*!\brief Construction and initialisation using the alignment configuration.
+     * \param[in] config The alignment configuration.
+     *
+     * \details
+     *
+     * Initialises the gap open score and gap extension score for this policy.
+     * If no gap cost model was provided by the user the default gap costs `-10` and `-1` are set for the gap open score
+     * and the gap extension score respectively.
+     */
+    explicit policy_affine_gap_recursion_banded(alignment_configuration_t const & config) : base_policy_t{config}
+    {}
+    //!\}
+
+    /*!\brief Initialises the first cell of a banded column that does not start in the first row of the matrix.
+     *
+     * \tparam affine_cell_t The type of the affine cell; must be an instance of seqan3::detail::affine_cell_proxy.
+     *
+     * \param[in] diagonal_score The previous diagonal score, which corresponds to \f$M[i - 1, j - 1]\f$.
+     * \param[in] previous_cell The predecessor cell corresponding to the value \f$H[i, j -1]\f$.
+     * \param[in] sequence_score The score obtained from the scoring scheme for the current cell (\f$ \delta\f$).
+     *
+     * \returns The computed affine cell.
+     *
+     * \details
+     *
+     * Computes the current cell according to following recursion formula:
+     * * \f$ H[i, j] = \max \{M[i, j - 1] + g_o, H[i, j - 1] + g_e\}\f$
+     * * \f$ M[i, j] = \max \{M[i - 1, j - 1] + \delta, H[i, j]\}\f$
+     */
+    template <typename affine_cell_t>
+    //!\cond
+        requires is_type_specialisation_of_v<affine_cell_t, affine_cell_proxy>
+    //!\endcond
+    affine_cell_type initialise_band_first_row_cell(score_type diagonal_score,
+                                                    affine_cell_t previous_cell,
+                                                    score_type const sequence_score) const noexcept
+    {
+        diagonal_score += sequence_score;
+        score_type horizontal_score = previous_cell.horizontal_score();
+
+        diagonal_score = (diagonal_score < horizontal_score) ? horizontal_score : diagonal_score;
```

Comment:
I guess this is because of simd, right? Instead of `std::min`.

We must find a good way to have a unified interface for

`std::min` and `seqan3::simd::min` or `int{5}` and `seqan3::simd::fill{5}` and so on.

So that we don't use weird constexpr if's in the code.

Reply:
Yes, especially since the ternary operator is only working with the gcc vector attributes.